### PR TITLE
[tr] Merging the `dev-tr` branch into main branch

### PR DESCRIPTION
**Pull request by aliok**
### Describe your changes

[In 8 months](https://github.com/cncf/glossary/issues/2123), the Turkish localization team hase accomplished to localize 20+ pages of content of the CNCF glossary to Turkish.

We made a lot of effort to make the content more understandable and accessible to the Turkish audience. 

We are looking forward to going on with our contributions (we already have 10 more documents that are translated and waiting for reviews!)

On behalf of the Turkish localization team, I would like to thank all the contributors and approvers who have helped us to achieve this milestone.
- Şeyma Demir @symys
- Uğur Yılmaz @uguryilmazdev
- Zehra Karakaya @0ZeKa0
- Muhsin Ozbek @muhsin09
- Enes Turan @devenes

We also would like to thank the approvers and emeritus approvers!
- @eminalemdar
- @halil-bugol
- @aliok
- @rwxdash
- @developer-guy

Thanks @seokho-son for your support!

### Checklist before opening this PR (put `x` in the checkboxes)
- [x] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [x] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -157,6 +157,14 @@ weight = 12
 [languages.zh-tw.params]
 description = "CNCF 雲端原生 Glossary 專案旨在用作討論雲端原生應用程式時常用術語的參考"
 
+[languages.tr]
+title = "Cloud Native Sözlüğü"
+description = "CNCF Cloud Native Sözlüğü projesi, cloud native uygulamalardan bahsederken kullanılan yaygın terimler için bir referans olarak kullanılmak üzere tasarlanmıştır."
+languageName ="Türkçe (Turkish)"
+contentDir = "content/tr"
+# Weight used for sorting.
+weight = 13
```
Comment (mboukhalfa) [inline config.toml:166]: I think this PR needs to be rebased because weight 13 already taken by Japenese lang on main

**Review by eminalemdar — APPROVED** [review]
LGTM!

**Comment by aliok** [comment]
Blocked by https://github.com/cncf/glossary/pull/2927/

**Comment by aliok** [comment]
@seokho-son this PR is ready now!

We're good to go for go-live, from our point of view.

**Comment by seokho-son** [comment]
Checklist to make localization go live:
- [x] Validate all files not in the localization directory (content/tr/*).
- [x] Ensure i18n/tr.toml exists.
- [x] Verify localization configuration in config.toml.
- [x] Acknowledge the initial list of localization contributors.
- [x] Ensure the Preview function works correctly.

**Comment by seokho-son** [comment]
LGTM!!!
A huge thank you to the Turkish localization team for your outstanding work!!
My apologies for the delayed response.

Let me merge this PR.

**Review by seokho-son — APPROVED** [review]

**Pull request merged**